src/ansys/fluent/core/fluent_connection.py

```diff
@@ -85,6 +93,42 @@ def __call__(self):
         return self._scheme_eval.scheme_eval("(data-valid?)")
 
 
+@dataclass(frozen=True)
+class FluentConnectionProperties:
+    """Stores Fluent connection properties, including connection IP, port and password;
+    Fluent Cortex working directory, process ID and hostname;
+    and whether Fluent was launched in a docker container.
+
+    Examples
+    --------
+    These properties are also available through the session object and can be accessed as:
+
+    >>> import ansys.fluent.core as pyfluent
+    >>> session = pyfluent.launch_fluent()
+    >>> session.connection_properties.list()
+    ['ip', 'port', 'password', 'cortex_pwd', 'cortex_pid', 'cortex_host', 'inside_container']
+    >>> session.connection_properties.ip
+    '127.0.0.1'
+    """
+
+    ip: str
+    port: int
+    password: str
+    cortex_pwd: str
+    cortex_pid: int
+    cortex_host: str
+    fluent_host_pid: int
+    inside_container: bool
+
+    def list_names(self) -> list:
+        """List all property names."""
+        return [k for k, _ in vars(self).items()]
+
+    def list_values(self) -> dict:
+        """Dict with all property names and values."""
+        return vars(self)
+
```

Comment:
I think the above 2 are same as `fields` and `asdict` methods of dataclass, we can mention them in the docstring.

Reply:
I don't see those using dir() 
![image](https://github.com/ansys/pyfluent/assets/132297401/0aa16960-3835-423e-89c8-335f546f7a01)
`__dataclass_fields__` isn't the same. `__dict__` returns the same as `list_values()`, I can make list_values a wrapper for `__dict__`, going to do that

Not finding anything particularly useful in dataclass documentation, just a separate `dataclasses.asdict()` https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict which seems to be very slow and does a lot more than we need here https://stackoverflow.com/questions/52229521/why-is-dataclasses-asdictobj-10x-slower-than-obj-dict

Reply:
I didn't realize those are not instance methods. Current methods are fine, thanks for checking.